### DISCUSSION TITLE
Li Ion Model With Known Capacity

### DISCUSSION THREAD

**Discussion by rob-g235** [pybamm-team/PyBaMM#2487] [Q&A]
Hi, 

I have a question about PyBaMM, and am also quite new to the depths of battery technology so excuse if my questions are slightly naive. But I'm not sure where to find this information.

Essentially I have a Li Ion battery from a manufacturer, that I'd like to model, in order to create a SOC (%) estimation algorithm for an embedded device.

So we have this battery, and the manufacturer has given us some specs, such as capacity, nominal voltage, and typical/max discharge current. I'm mainly interested in getting a voltage discharge curve when it's discharged at ~400uA.

When specifying parameters I thought you could just state 'Nominal Cell Capacity', but from reading another thread this doesn't actually model the capacity of the battery, and instead I need these paramteters:

- 'Electrode height [m]'
- 'Electrode width [m]'
- 'Negative electrode thickness [m]'
- 'Positive electrode thickness [m]'
- 'Negative electrode active material volume fraction'
- 'Positive electrode active material volume fraction'

So my question is, is it possible to get a reasonable guess of these parameters based off the actual physical size of the battery itself? I know from a real life test it lasts about 5 hrs when 7mA is drawn, so if the model could acheive this I could at least know it's reasonable?

**Comment by RuiheLi** [comment]
Hi, unfortunately, I think the answer is no. There are quite a lot of parameters that are required for the DFN model in PyBaMM. Other than the ones you mentioned, the OCV curves of both electrodes are also very important. [This](https://iopscience.iop.org/article/10.1149/1945-7111/ab9050/meta) paper gives very comprehensive explanation about parametrising the DFN model on a LIB cell. I think if you cannot open the cell for these complex characterization tests, the alternative may be to parameterise an equivalent circuit model. I am not very familiar with this. But I believe you can easily find some papers on this.

**Comment by valentinsulzer** [comment]
Hi @rob-g235, finding parameters like this isn't currently within the scope of the open-source software, but if you email me (valentin.sulzer@gmail.com) with more details I might be able to help

**Comment by rob-g235** [comment]
@RuiheLi Hmm fair enough. I don't think we have the time, or resources to open up the cell. I assume all the other models in PyBaMM are the same, as in they require a multitude of parameters beyond the basic ones I have from the manufacturer, to even get a reasonable model.

@tinosulzer Sure thing, thanks very much, I'll shoot you over an email shortly.

**Comment by objectivezhou** [comment]
Got,Thankyou!
